src/phpDocumentor/ConfigurationFactory/PhpDocumentor3.php

```diff
@@ -0,0 +1,176 @@
+        $ignoreSymlinks = filter_var($version->api->ignore->attributes()->symlinks, FILTER_VALIDATE_BOOLEAN);
+
+        return [
+            'folder' => (string) $version->folder,
+            'api'    => [
+                'format'               => ((string) $version->api->attributes()->format) ?: 'php',
+                'source'               => [
+                    'dsn'   => (string) $version->api->source->attributes()->dsn,
+                    'paths' => ((array) $version->api->source->path) ?: ['src'],
+                ],
+                'ignore'               => [
+                    'hidden'   => $ignoreHidden,
+                    'symlinks' => $ignoreSymlinks,
+                    'paths'    => ((array) $version->api->ignore->path) ?: [],
+                ],
+                'extensions'           => $extensions,
+                'visibility'           => (string) $version->api->visibility,
+                'default-package-name' => ((string) $version->api->{'default-package-name'}) ?: 'Default',
+                'markers'              => (array) $version->api->markers->children()->marker,
+            ],
+            'guide'  => [
+                'format' => ((string) $version->guide->attributes()->format) ?: 'rst',
+                'source' => [
+                    'dsn'   => ((string) $version->guide->source->attributes()->dsn) ?: 'file://.',
+                    'paths' => (array) $version->guide->source->path,
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * Builds the template part of the array from the phpDocumentor3 configuration xml.
+     *
+     * @param \SimpleXMLElement $template
+     *
+     * @return array
+     */
+    private function buildTemplate(\SimpleXMLElement $template)
+    {
+        if (!$template) {
+            // Use default template if none is found in the configuration
+            return [
+                [
+                    'name' => 'clean',
+                ],
+            ];
+        }
+
+        $array = [];
+        foreach ($template->attributes() as $key => $value) {
+            $array[] = [
+                (string) $key => (string) $value,
+            ];
+        }
+
+        return $array;
+    }
+
+    /**
+     * Validates the phpDocumentor3 xml structure against the schema defined in the schemaPath.
+     *
+     * @param $phpDocumentor
+     *
+     * @throws \InvalidArgumentException if the xml structure is not valid.
+     */
+    private function validateXmlStructure(\SimpleXMLElement $phpDocumentor)
+    {
+        libxml_clear_errors();
+        libxml_use_internal_errors(true);
+
+        $dom        = new \DOMDocument();
+        $domElement = dom_import_simplexml($phpDocumentor);
+        $domElement = $dom->importNode($domElement, true);
+        $dom->appendChild($domElement);
+
+        $dom->schemaValidate($this->schemaPath);
+
+        $error = libxml_get_last_error();
+
+        if ($error) {
```

Comment:
Please make this check more explicit like $error !== false since libxml_get_last_error will return false when no error occurred.